src/browserlib/extract-dfns.mjs

```diff
@@ -225,6 +228,173 @@ export default function (spec, idToHeading = {}) {
+        dfnName = dfnName.replace(/.*\.prototype\./, '');
+
+        dfn.dataset.lt = dfnName;
+        // set dfn-type
+        if (dfn.dataset.lt.match(/\(/)) {
+          dfnName = cleanMethodName(dfnName);
+          dfn.dataset.lt = dfnName;
+          dfn.dataset.dfnType = "method";
+        } else {
+          dfn.dataset.dfnType = "attribute";
+        }
+      } else {
+        // We handle other headings that look like a method / property
+        // on an object instance (rather than its prototype)
+        // or an abstract op
+
+        // if there is already a dfn element, we move on
+        if (el.querySelector("dfn")) return;
+
+        // only dealing with well-known patterns
+        if (!dfnName.match(/^[a-z]+\.[a-z]+/i) // à la JSON.parse
+            && !dfnName.match(/^([A-Z][a-z]+)+ *\(/) // à la ArrayCreate (
+           ) return;
+        // Skip symbol-based property definitions
+        if (dfnName.match(/@@/)) return;
+
+        // Skip headings where foo.bar appears as part of a longer phrase
+        if (!dfnName.match(/\(/) && dfnName.match(/ /)) return;
+
+        // redundant definitions of constructors on the global object
+        // e.g. "Array ( . . . )"
+        if (dfnName.match(/\. \. \./)) return;
+
+        const dfn = wrapWithDfn(el);
+
+        if (dfnName.match(/^[a-z]+\.[a-z]+/i)) {
+          // set definition scope
+          // This assumes that such methods and attributes are only defined
+          // one-level deep from the global scope
+          dfn.dataset.dfnFor = dfnName.replace(/\..*$/, '');
+          // FIXME Math.hypot
+          dfnName = dfnName.replace(dfn.dataset.dfnFor + ".", '');
+          if (dfnName.match(/\(/)) {
+            dfnName = cleanMethodName(dfnName);
+            dfn.dataset.lt = dfnName;
+            dfn.dataset.dfnType = "method";
+          } else {
+            dfn.dataset.lt = dfnName;
+            dfn.dataset.dfnType = "attribute";
+          }
+        } else { // Abstract ops à la ArrayCreate or global constructor
+          dfnName = cleanMethodName(dfnName);
+          dfn.dataset.lt = dfnName;
+          // distinguish global constructors from abstract operations
+          if ((dfn.closest("emu-clause")?.parentNode?.id || "").match(/-constructors?$/)) {
+            dfn.dataset.dfnType = "constructor";
+          } else {
+            dfn.dataset.dfnType = "abstract-op";
+          }
+        }
+        definitionNames.add(dfnName);
+      }
+    });
+
+  [...document.querySelectorAll(`${sectionFilter} dfn`)]
+    .forEach(el => {
+      // Skip definitions in conformance page and conventions page
+      if (el.closest('section[data-reffy-page$="conformance.html"]') ||
+         el.closest('section[data-reffy-page$="notational-conventions.html"]')) {
+        return;
+      }
+      // If the <dfn> has no id, we attach it the one from the closest
+      // <emu-clause> with an id
+      // Note that this means several definitions can share the same id
```

Comment:
Does that happen in practice?

This makes me wonder whether there exist cases in other specs where different definitions share the same ID. Maybe that already happens with some of the dfns that are in headings. Do we want to end up in a situation where there is only one entry per ID in dfn extracts or can we simply not care?

Reply:
yes, it does happen in practice - the 1190 extracted definitions only expose 1103 distinct ids.

But it seems to be already the case in our existing set of definitions:
```bash
$ jq '.dfns[].href' *.json|wc -l
33911
$ jq '.dfns[].href' *.json|sort|uniq|wc -l
33288
```

I'm also not aware that anything in the definition ecosystem relies on the unicity of hrefs, so I'm not particularly bothered at this point.